pkg/api/networkservice/mechanisms/vfio/helpers.go

```diff
@@ -17,51 +18,124 @@
 package vfio
 
 import (
+	"strconv"
+
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
-	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/common"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/cls"
 )
 
-// Mechanism - vfio mechanism helper
-type Mechanism interface {
-	GetNetNSInode() string
-	GetParameters() map[string]string
-	GetPCIAddress() string
+// Mechanism is a vfio mechanism helper
+type Mechanism struct {
+	*networkservice.Mechanism
 }
 
-type mechanism struct {
-	*networkservice.Mechanism
+// New returns *networkservice.Mechanism of type vfio with the given cgroupDir
+func New(cgroupDir string) *networkservice.Mechanism {
+	return &networkservice.Mechanism{
+		Cls:  cls.LOCAL,
+		Type: MECHANISM,
+		Parameters: map[string]string{
+			CgroupDirKey: cgroupDir,
+		},
+	}
 }
 
 // ToMechanism converts unified mechanism to helper
-func ToMechanism(m *networkservice.Mechanism) Mechanism {
+func ToMechanism(m *networkservice.Mechanism) *Mechanism {
 	if m.GetType() == MECHANISM {
-		return &mechanism{
+		return &Mechanism{
 			m,
 		}
 	}
 	return nil
 }
 
-// GetParameters returns map of mechanism parameters
-func (m *mechanism) GetParameters() map[string]string {
-	if m == nil {
-		return nil
+// GetParameters returns the map of all parameters to the mechanism
+func (m *Mechanism) GetParameters() map[string]string {
+	if m.Parameters == nil {
+		m.Parameters = map[string]string{}
 	}
 	return m.Parameters
 }
 
-// GetParameters returns network namespace Inode
-func (m *mechanism) GetNetNSInode() string {
-	if m == nil || m.GetParameters() == nil {
-		return ""
-	}
-	return m.GetParameters()[common.NetNSInodeKey]
+// GetCgroupDir returns client on host cgroup directory
+func (m *Mechanism) GetCgroupDir() string {
+	return m.GetParameters()[CgroupDirKey]
+}
+
+// SetCgroupDir sets client on host cgroup directory
+func (m *Mechanism) SetCgroupDir(cgroupDir string) {
+	m.Parameters[CgroupDirKey] = cgroupDir
```

Comment:
Won't this panic if m == nil or GetParameters() == nil?

Reply:
Fixed, now GetParameters() returns empty map for m == nil.